internal/cli/atlas/streams/connection/create.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This code was autogenerated at 2023-07-05T01:21:22Z. Note: Manual updates are allowed, but may be overwritten.
+
+package connection
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/file"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20230201004/admin"
+)
+
+type CreateOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	store           store.ConnectionCreator
+	name            string
+	filename        string
+	streamsInstance string
+	fs              afero.Fs
+}
+
+func (opts *CreateOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+var createTemplate = "Connection {{.Name}} created.\n"
+
+func (opts *CreateOpts) Run() error {
+	createRequest, err := opts.newCreateRequest()
+	if err != nil {
+		return err
+	}
+
+	r, err := opts.store.CreateConnection(opts.ConfigProjectID(), opts.streamsInstance, createRequest)
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+func (opts *CreateOpts) newCreateRequest() (*atlasv2.StreamsConnection, error) {
+	out := atlasv2.NewStreamsConnectionWithDefaults()
+	if err := file.Load(opts.fs, opts.filename, out); err != nil {
+		return nil, err
+	}
+	out.Name = &opts.name
```

Comment:
can the name be defined on the file? for clusters for example we make the name arg optional to allow setting it on the file

Reply:
Make this optional in 18902c69